### PR TITLE
crimson/osd: implement flush_pg_stats command

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/osd/heartbeat.cc [hunk 1]
```diff
@@ -184,9 +184,19 @@ void Heartbeat::update_peers(int whoami)
...
+  for (auto& peer : peers) {
+    osds.push_back(peer.first);
+  }
+  return osds;
+}
+
 void Heartbeat::remove_peer(osd_id_t peer)
 {
-  auto found = peers.find(peer);
+  [[maybe_unused]] auto found = peers.find(peer);
```
Comment (rzarzynski) [inline src/crimson/osd/heartbeat.cc:199]: `peers` is an instance of `std::map<osd_id_t, Peer>`, so maybe let's exchange the current assert with:
```cpp
  assert(peers.count(peer) == 1);
```

File: src/crimson/mgr/client.h [hunk 2]
```diff
@@ -24,9 +24,8 @@ namespace crimson::mgr
 // implement WithStats if you want to report stats to mgr periodically
 class WithStats {
 public:
-  // the method is not const, because the class sending stats might need to
-  // update a seq number every time it collects the stats
-  virtual MessageRef get_stats() = 0;
+  virtual void update_stats() = 0;
+  virtual MessageRef get_stats() const = 0;
```
Comment (rzarzynski) [inline src/crimson/mgr/client.h:28]: :+1:

File: src/crimson/osd/heartbeat.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/osd/heartbeat.cc]: @rzarzynski updated and repushed.

**Comment by tchaikov** [comment]
```
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.176467/0.16582  => accepted
09:34:53 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: cpu_cycles_per_op: (or (less) (near 0.05)):: 1558599.2291666667/1413416.359375  => rejected
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 45.0/42.0  => accepted
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 2.00)):: 9.2376/17.6163  => accepted
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.354108/0.376838  => accepted
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.176006/0.159158  => accepted
09:34:53 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: cpu_cycles_per_op: (or (less) (near 0.05)):: 1547850.2689655172/1402459.6434108526  => rejected
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 45.0/40.0  => accepted
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 2.00)):: 9.53939/17.8979  => accepted
09:34:53 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.35128/0.375064  => accepted
09:34:53 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 56.0866/82.5466  => rejected
09:34:53 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: cpu_cycles_per_op: (or (less) (near 0.05)):: 307911.6179236973/233299.45393407522  => rejected
09:34:53 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 14358.0/21131.0  => rejected
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 2.00)):: 19538.8/22893.8  => accepted
09:34:53 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.00111024/0.000753383  => rejected
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 68.492/58.362  => accepted
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: cpu_cycles_per_op: (or (less) (near 0.05)):: 252141.86893839613/329980.1723130734  => accepted
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 17533.0/14940.0  => accepted
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 2.00)):: 20255.8/18919.2  => accepted
09:34:53 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.000908576/0.00106675  => accepted
09:34:53 - WARNING  - cbt      - 6 tests failed out of 20
```

**Pull request merged**